src/graph/graph.c

```diff
@@ -1167,6 +1141,78 @@ void Graph_BulkDelete(Graph *g, Node *nodes, uint node_count, Edge *edges, uint
 	if(edge_deleted != NULL) *edge_deleted = _edge_deleted;
 }
 
+int Graph_UpdateNode
+(
+	Graph *g,
+	Node *node,
+	Attribute_ID attr_id,
+	SIValue new_value
+) {
+	ASSERT(g);
+	ASSERT(node);
+
+	int res         = 0;
+	GraphEntity *ge = (GraphEntity *)node;
+
+	// handle the case in which we are deleting all properties
+	if(attr_id == ATTRIBUTE_ALL) return GraphEntity_ClearProperties(ge);
+
+	// try to get current property value
+	SIValue *old_value = GraphEntity_GetProperty(ge, attr_id);
+	
+	g->CrudHub.NodeUpdated(g, node, attr_id, old_value);
+
+	if(old_value == PROPERTY_NOTFOUND) {
+		// adding a new property; do nothing if its value is NULL
+		if(SI_TYPE(new_value) != T_NULL) {
+			res = GraphEntity_AddProperty(ge, attr_id, new_value);
+		}
+	} else {
+		// update property
+		res = GraphEntity_SetProperty(ge, attr_id, new_value);
+	}
+
+	SIValue_Free(new_value);
```

Comment:
Please create a third shared function `Graph_UpdateEntity` which both `Graph_UpdateNode` and `Graph_UpdateEdge` will call.

OR Reduce both `Graph_UpdateNode` and `Graph_UpdateEdge` into just `Graph_UpdateEntity` which will accept the entity type.